src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3041,10 +3072,20 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
             startingState.ForEach(
                 (slot, variables) =>
                 {
-                    var symbol = variables[variables.RootSlot(slot)].Symbol;
-                    if (Symbol.IsCaptured(symbol, localFunc))
+                    if (reachable)
                     {
-                        SetState(ref state, slot, GetState(ref startingState, slot));
+                        var symbol = variables[variables.RootSlot(slot)].Symbol;
+                        if (Symbol.IsCaptured(symbol, localFunc))
+                        {
+                            SetState(ref state, slot, GetState(ref startingState, slot));
+                        }
+                    }
+                    else if (variables[slot].ContainingSlot == 0)
```

Comment:
>variables[slot].ContainingSlot == 0 [](http://example.com/codeflow?start=29&length=35)

Since we already have a ```Symbol.IsCaptured(symbol, localFunc)```check above, and it looks pretty trivial, we should consider using it here as well.